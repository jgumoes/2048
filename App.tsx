import { StatusBar } from 'expo-status-bar';
import { ErrorBoundary } from "react-error-boundary";
import { SafeAreaView } from 'react-native';
import Grid, { colorTestGrid } from './src/grid';
import React, { useState } from 'react';
import { GestureHandlerRootView } from 'react-native-gesture-handler';
import { appStyles } from './AppStyles';
import GameBoard from './components/GameBoard';

const swipeForGameWonGrid = () => [
  [ 0, 0, 0, 0], 
  [ 0, 1024, 1024, 0],
  [ 0, 1024, 1024, 0], 
  [ 0, 0, 0, 0], 
]
export default function App() {
  // const [grid, setGrid] = useState(new Grid({gridSize: 4}))
  // const [grid, setGrid] = useState(new Grid({grid: colorTestGrid(), undoCount: 5}))
  const [grid] = useState(() => {
    const newGrid = new Grid({grid: swipeForGameWonGrid()})
    // const newGrid = new Grid({grid: colorTestGrid(), score: 42069, undoCount: 5})
    // const newGrid = new Grid({grid: [
    //   [ 16, 8, 4, 2], 
    //   [ 2, 4, 2, 4], 
    //   [ 4, 2, 4, 2], 
    //   [ 32, 4, 2, 4]
    // ]})
    // GridOverlayState.primeGameWon = newGrid.isGameWon
    return newGrid
  })
  
  return (
    <SafeAreaView style={[appStyles.container]}>
      <StatusBar style="auto" hidden={false} />
      <GestureHandlerRootView style={[appStyles.container, {marginTop: 80}]}>
        <GameBoard grid={grid}/>
      </GestureHandlerRootView>
    </SafeAreaView>
  );
}
