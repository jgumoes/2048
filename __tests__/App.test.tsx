import { render, screen } from "@testing-library/react-native"
import App from "../App"

jest.mock('expo-status-bar', ()=>{
  return({
    StatusBar: () => {
      <view />
    }
  })
})

jest.mock('react-native-gesture-handler', ()=>{
  return({
    GestureHandlerRootView: () => {
      <view />
    },
    gestureHandlerRootHOC: (element: JSX.Element) => element
  })
})

test('App renders', () => {
  render(<App />)
  expect(screen.toJSON()).not.toBeNull()
  expect(screen).toMatchSnapshot()
})