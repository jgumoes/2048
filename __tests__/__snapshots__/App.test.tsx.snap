// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders 1`] = `
<RCTSafeAreaView
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "rgb(251, 248, 239)",
        "flex": 1,
        "justifyContent": "center",
      },
    ]
  }
/>
`;
